### ISSUE TITLE
🐛 index out of bounds panic in module_resolver.rs when checking Effect.map(...) call (recurrence of #7355/#7368)

### ISSUE THREAD

**Issue by erick-winston** [biomejs/biome#10938]
### Environment information

```
CLI:
  Version:                      2.5.3

Platform:
  OS:                           macOS (darwin arm64)

Environment:
  JS_RUNTIME_NAME:               bun
  JS_RUNTIME_VERSION:             1.3.5
```

### What happened?

`biome check` (via `ultracite check`, which wraps biome 2.5.3) panics with an `index out of bounds` error in `crates/biome_module_graph/src/js_module_info/module_resolver.rs:500:33` when checking a file that calls `Effect.map(...)` (from the [`effect`](https://www.npmjs.com/package/effect) package) — either via `.pipe(Effect.map(...))` or the direct two-argument form `Effect.map(someEffect, fn)`.

This looks like a recurrence of #7355 and #7368, both of which report the exact same panic signature (`module_resolver.rs`, "index out of bounds") and were auto-closed for inactivity/no-repro rather than fixed. I hit this independently on a real monorepo file and was able to reduce it to a minimal, dependency-light repro.

### Minimal reproduction

```ts
// repro.ts
import { Effect } from "effect";

const result = Effect.map(Effect.succeed(1), (n) => n + 1);

export { result };
```

```
bun add effect
bunx @biomejs/biome check repro.ts
```

Output:

```
Biome encountered an unexpected error

Source Location: crates/biome_module_graph/src/js_module_info/module_resolver.rs:500:33
Thread Name: biome::workspace_worker_0
Message: index out of bounds: the len is 9 but the index is 1915

repro.ts internalError/panic  INTERNAL

  × processing panicked: index out of bounds: the len is 9 but the index is 1915
```

Notes from bisecting:
- `Effect.succeed(1)` alone (no `.map`) does **not** panic.
- `Effect.map(someEffect, callback)` panics regardless of whether it's written via `.pipe(Effect.map(...))` or the direct two-arg form.
- The reported `len` varies with how many other imports/exports are in the file (9 in the minimal case above), but the `index` is always `1915`, suggesting something is using a byte offset (or similar large computed value) as an index into a small resolved-imports/module-graph array rather than bounds-checking it.
- Confirmed on a real file (an Effect-based Arcjet security middleware) using a longer `.pipe(Effect.provide(...), Effect.map(...), Effect.catchTag(...))` chain before reducing to the snippet above — so this isn't specific to any one downstream package, just `Effect.map`'s shape.

### Expected result

`biome check` should either check the file normally or return a normal diagnostic — not panic.

### Related

- #7355 (same panic signature, closed for inactivity without a fix)
- #7368 (same panic signature, closed for inactivity without a fix)

Since both prior reports were closed only due to missing a minimal reproduction (not because the underlying bug was fixed), hopefully this smaller repro is enough to actually track it down.
